### PR TITLE
Add update calculations

### PR DESCRIPTION
**Pull request by coderkevin**
This adds update calculations based on requirements and the current
time.

To Test:
1. `npm test` and ensure all tests pass.

### CODE REVIEW

**Comment by coderkevin** [comment]
Adjusting how requirements are kept in order to be more flexible with API queries.

File: src/client/__tests__/calculate-updates.spec.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+				endpoints: {
+					1: { lastReceived: now - ( 40 * SECOND ) },
+				},
+				queries: [
+					{ params: { page: 1, perPage: 3 }, lastReceived: now - ( 40 * SECOND ) },
+				],
+			},
+		};
+		const { nextUpdate } = calculateUpdates( requirementsByEndpoint, endpointsState );
+		expect( nextUpdate / 1000 ).toBeCloseTo( ( 20 * SECOND ) / 1000, 1 );
```
Comment (timmyc) [inline src/client/__tests__/calculate-updates.spec.js]: TIL `toBeCloseTo`

File: src/client/calculate-updates.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	requirementsByEndpoint,
+	endpointsState,
+	now = new Date()
+) {
+	const updateInfo = { updates: [], nextUpdate: DEFAULT_NEXT_UPDATE };
+	appendUpdatesForEndpoints( updateInfo, [], requirementsByEndpoint, endpointsState, now );
+	return updateInfo;
+}
+
+function appendUpdatesForEndpoints( updateInfo, endpointPath, requirementsByEndpoint, endpointsState, now ) {
```
Comment (timmyc) [inline src/client/calculate-updates.js]: Do we want docblocks on these methods?

File: src/client/__tests__/calculate-updates.spec.js (hunk 1 above)
Comment (coderkevin) [reply src/client/__tests__/calculate-updates.spec.js]: Yeah, these tests were giving me fits as the date math was barely imprecise. Probably using floating point math internally even though I'm dealing in integers. That's why I did the `/ 1000`.

I figure if they're accurate to < 1 second, that's good enough for our purposes here.

File: src/client/calculate-updates.js (hunk 2 above)
Comment (coderkevin) [reply src/client/calculate-updates.js]: Yeah, I suppose. They're not public, but some extra explanation can't hurt. I'll update and commit. Thanks!

**Pull request merged**